### DISCUSSION TITLE
Meeting: Celo Wallet and (d)App Council #23 [May] 💬

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/celo-monorepo#10308] [Wallet and (d)App Council]
# Monthly Celo Wallet and (d)App Council

##  📆 Quick links 

- When: **Wednesday, May 10 (9am PT/6pm CET)** 
- Where: **[Google Meet](https://meet.google.com/uwj-wfse-uzm)**
- Agenda: **comments**👇
- Previous: https://github.com/celo-org/celo-monorepo/discussions/10248
- Next: after meeting
- Recording: after meeting

## 🤷‍♂️ What is this?

A monthly meeting for Celo **developers, founders,** **product** people (anyone really!) to come together and
- find consensus (coordinate),
- learn about needs, pain points, and desires
- present and collect technical feedback

## 🗒️  Agenda

- Comment with a short sentence describing what you'd like to discuss
- Don't worry, it can be small or big! 
- Every builder is welcome (and we are very easy going, I promise 😄)
- Upvote topics you're interested in

## Housekeeping

_You can safely ignore this!_

- [ ] Send update calendar invite with Github link
- [ ] Encourage additions to the agenda
- [ ] Publicise on forum and Discord
- [ ] RECORD meeting 🎥 
- [ ] Share recording below and on forum
- [ ] Create next meeting (link to next meeting above)

**Comment by arthurgousset** [comment]
[open discussion] Does anyone have experience using [viem](https://viem.sh/)?

**Reply by arthurgousset** [reply]
Meeting notes:
- Jean 
  - started looking at it to replace contractkit
  - Looked at it to use it in a backend project to get data from the blockchain
  - Jacob looked into it to replace contractkit but seemed to miss account management features
  - Could replace ethers in the long-term
  - Also saw an official announcement that Wagmi is using viem under the hood
- Charlie
  - With ethers you need to use Typechain to get typed contracts. With viem you get statically typed method calls out of the box.
  - Typechain is like a plugin with Ethers. You add a build script that runs a typechain command which generates TS types from the ABI.
  - Typechain folks don't recommend publishing TS types and recommend generating them instead
  - viem infers types from JSON ABIs.
  - It's v0 though (expect breaking changes)
 
Ethers usage in Valora: 
- resolve-kit
- fiat-connect client
- branch with work going on to replace ethers with contractkit (draft PR with 
  - https://github.com/valora-inc/wallet/pull/3621
- Ask: keep us aware with painpoints (maybe turn migration from contractkit to Ethers v5 into a DevRel tutorial)

**Reply by jxom** [reply]
What account features are missing? viem supports both JSON-RPC Accounts & Local Accounts (Private Key & HD: mnemonic, etc).

We don't expect any large breaking changes in viem in the near future as well! We are hoping to release viem v1 very soon so consumers know we have settled on our APIs.

**Reply by jeanregisser** [reply]
Hey, this was pre viem 0.2. We haven't tried that part again since it was added.
Great to hear about viem reaching a stability milestone very soon and congrats 🎉 
I kinda guessed this was coming, following the announcement of wagmi using viem.

**Reply by arthurgousset** [reply]
Thanks for chiming in here @jxom ! Appreciate the updates.
Out of curiosity, can I ask how you found this Github Discussion? 
Do you use some form of "viem" search across Github?

**Reply by jxom** [reply]
Lol yep! 😂

**Reply by arthurgousset** [reply]
Love it 😄 @jxom what's the tool (how do you do it) if you don't mind sharing?
Very interested in doing something similar!

**Reply by jxom** [reply]
Just GitHub search at the top of page!

**Reply by arthurgousset** [reply]
Ah, that makes sense :) I recently saw that Github had "[advanced search](https://github.com/search/advanced)" and a new [Github search API ](https://docs.github.com/en/rest/search?apiVersion=2022-11-28) so I thought it might be something fancy like that.

**Comment by arthurgousset** [comment]
[update] Native safe launch at [app.safe.global](https://app.safe.global/)

[[Multisig] Native Safe launch (migration guidance)](https://forum.celo.org/t/multisig-native-safe-launch-migration-guidance/5705)

**Reply by arthurgousset** [reply]
Meeting notes: 

- Jean was interested in adding Safe in the dapplist in Valora (there is an error message that warns that only Metamask is supported, when actually WalletConnect works well).
- Next step: Connect Jean with uxiof at Safe.

**Comment by arthurgousset** [comment]
[releases]

+ [@celo/contracts](https://www.npmjs.com/package/@celo/contracts) has source code of all core contracts
+ new [@celo/abis](https://www.npmjs.com/package/@celo/abis) has ABI (JSON) and TS types for core contracts
+ [@celo/contractkit](https://www.npmjs.com/package/@celo/contractkit) has new core contracts release 9 changes

**Reply by arthurgousset** [reply]
Meeting notes:

- Charlie: 
  - can be tricky to use generic EVM libraries on Celo because they don't support fee currency by default
  - Charlie set up automatic releases using [semantic-release](https://github.com/semantic-release/semantic-release) in Valora (used it in open source fiatconnect repo). It's been very useful because there are more community contributions. Also use renovate set up which manages downstream dependencies.
- Jean:
  - Sometimes long delay between PRs merged and NPM releases. For example with contractkit changes.
 - [x] Ask: Import only specific ABIs instead of importing all ABIs.

**Comment by arthurgousset** [comment]
Discussion about usefulness of the meeting:

+ generally very useful re being in the loop about what happens at cLabs, hear about concerns from others. Maybe we need to **advertise** it more. 
+ we need a regular touch point, but we could scale it up/down depending on need, maybe we need to do more **advertising**. It's like the coffee machine meeting.
+ sending out an **agenda** in advance could be useful ("Hey, we are talking about these things, join if you're interested"). If you attend and don't know what you get out of it.

**Comment by fbaud** [comment]
Getting some feedback on upcoming account abstraction for EVM blockchains (ref: https://docs.celo.org/blog/tutorials/erc-4337-overview-protocol#goals-and-benefits-of-erc-4337)

**Reply by arthurgousset** [reply]
Meeting notes:

- Jean: Valora used [MetaTransactionWallet.sol](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/common/MetaTransactionWallet.sol) to automate decentralised phone number verification in the early days. It came with a lot of complexity and has been removed slowly over time. This was very use case specific.
- Arthur: [Safe {core}](https://safe.global/core) has very interesting AA SDKs
- Frederic: Interesting "gas bootstrap" problem that AA might solve with **relayers**. E.g. receiving cUSD but not having CELO for gas.
